dashdblocal_notebooks/livy-notebooks/IoT_Demo_Pyspark.ipynb

```diff
@@ -0,0 +1,1246 @@
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Kafka assembly found"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "\n",
+    "path = \"/mnt/bludata0/home/bluadmin/spark/defaultlibs\"\n",
+    "kafka_assembly = \"spark-streaming-kafka-0-8-assembly_2.11-2.3.0.jar\"\n",
+    "url = \"http://repo1.maven.org/maven2/org/apache/spark/spark-streaming-kafka-0-8-assembly_2.11/2.3.0/\" + kafka_assembly\n",
+    "\n",
+    "if (os.path.exists(path + \"/\" + kafka_assembly)):\n",
+    "    print(\"Kafka assembly found\")\n",
+    "else:\n",
+    "    os.environ['PYSPARK_SUBMIT_ARGS'] = '--packages org.apache.spark:spark-streaming-kafka-0-8_2.11:2.3.0 pyspark-shell'\n",
+    "    print (\"Please download the Kafka assembly from the following link: \"+url+\" and restart the notebook\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Kafka configuration to connect to our IoT demo data producer__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we define a utility function to set up a Kafka Streaming context, and run it for an optionally defined period of time with a given processing function. \n",
+    "\n",
+    "**Important note** : When we run this interactively in a notebook cell, it's important to set the timeout to a specific value to stop the streaming context. Otherwise it will keep running and produce cell output, even after the cell has finished executing. This timeout is defined in ssc.awaitTerminationOrTimeout().\n",
+    "\n",
+    "Note also that a streaming context cannot be re-used after it has executed, so we can't keep it across cells and therefore we have to use the utility function to re-create it before every execution.\n",
+    "\n",
+    "We use a *finally* block to make sure the context is stopped even in case of exception; otherwise the whole streaming infrastructure becomes unusable within the notebook kernel's lifetime."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "topics=[\"iot4dashdb\"]\n",
+    "\n",
+    "def runStreamingContext(rdd):\n",
+    "    sc = SparkContext.getOrCreate()\n",
```

Comment:
Nitpicking: Both `sc` and `sqlsc` are already created two cells earlier. Do they need to be recreated here or can we re-use the ones from earlier?

Reply:
yes it's needed, I have tested without and it doesn't work